packages/@aws-cdk/cdk/lib/cloudformation/cross-account-region-token.ts

```diff
@@ -0,0 +1,42 @@
+import { IConstruct, ResolveContext, Stack } from '..';
+import { Token } from '../core/tokens';
+
+export abstract class CrossAccountRegionToken extends Token {
+  private readonly owningStack: Stack;
+
+  /**
+   * @param regularValue the value used when this is referenced NOT from a cross account and/or region Stack
+   * @param crossAccountRegionValue the value used when this is referenced from a cross account and/or region Stack
+   * @param scope the scope this reference is mastered in. Used to determine the owning Stack
+   * @param displayName a short name to be used in Token display
+   */
+  protected constructor(private readonly regularValue: any, private readonly crossAccountRegionValue: any,
```

Comment:
I don't want people to have to interact with these tokens directly (see my comment about `cdk.AmazonResourceName`), they should be private classes.

1. Use a "props" interface instead of multiple arguments (rule of thumb should be >2 arguments => props).
2. Why are `regularValue` and `crossAccountRegionValue` of type `any` and not `string`?
3. Terminology: instead of cross-account-region, how about shorten it to cross-env ("env" in the CDK is account+region pair)

Reply:
Sure. Will change as part of the `cdk.AmazonResourceName`.

Reply:
Changed.